v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/iowrapper/JdbcDataSource.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.source.reader.io.jdbc.iowrapper;
+
+import com.google.cloud.teleport.v2.source.reader.auth.dbauth.DbAuth;
+import com.google.cloud.teleport.v2.source.reader.io.jdbc.iowrapper.config.JdbcIOWrapperConfig;
+import com.google.cloud.teleport.v2.spanner.migrations.utils.JarFileReader;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.Serializable;
+import java.net.URLClassLoader;
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.apache.commons.lang3.StringUtils;
+import org.jline.utils.Log;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class JdbcDataSource extends BasicDataSource implements Serializable {
+
+  /*
+   * Implementation Detail, we take required members of JDBCIOWrapperConfig as private members here since there are members of JDBCIOWrapperConfig like FluentBackoff which aren't marked serializable by Apache Beam.
+   */
+
+  private final String sourceDbURL;
+  private final DbAuth dbAuth;
+  private final ImmutableList<String> initSql;
+  private final Long maxConnections;
+  private final String jdbcDriverJars;
+  private final String jdbcDriverClassName;
+  private static final Logger LOG = LoggerFactory.getLogger(JdbcDataSource.class);
+
+  /**
+   * Sets the {@code testOnBorrow} property. This property determines whether or not the pool will
```

Comment:
link to the DBCP parameters and docs here ?

Reply:
Thanks.
Added at the top of the class. Also added a detailed reasoning for this class in the javadocs.